### DISCUSSION TITLE
about esbuild charset

### DISCUSSION THREAD

**Discussion by xxleyi** [johnlindquist/kit#867] [General]
from https://esbuild.github.io/api/#charset
get:
 > By default esbuild's output is ASCII-only. Any non-ASCII characters are escaped using backslash escape sequences. One reason is because non-ASCII characters are misinterpreted by the browser by default, which causes confusion.

As for kit, maybe it is better to use `utf-8` as default charset value?

**Comment by xxleyi** [comment]
I have a reason to notice this thing. Now I have some scripts, which will use applescript execute javascript in browser. Sometimes, there are some Chinese character, which will make my script not work as expected.

**Reply by johnlindquist** [reply]
Are you using TypeScript scripts?

Could you try editing:

~/.kit/api/kit.js

And search for "buildTSScript" then you can update the esbuild options in that function (I'm on my phone).

![image](https://user-images.githubusercontent.com/36073/185724620-96cbac85-431a-4788-9a28-f10f0a4f4e94.png)

And confirm that fixes it for you?

> !!! You will have to hit the main Kit.app shortcut 3 times to clear the process cache after editing the SDK files


> !!! If anything breaks when editing those files, quit Kit.app,  delete the entire "~/.kit" directory, then opening Kit.app again will re-install the ~/.kit directory

**Reply by xxleyi** [reply]
yeah, I have changed to utf-8 and it worked

**Reply by johnlindquist** [reply]
OK, awesome. I'll run some tests with it to make sure it doesn't break anything and try to get it into the next build.

If you ever want to chat about any  script ideas/features we're starting up a Discord here:

https://discord.gg/naxcYT9A
